### ISSUE TITLE
textarea aria-describedby references span before it is present

### ISSUE THREAD

**Issue by benstoltz** [Esri/calcite-design-system#11633]
### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates and reviewed the [Accessibility page](https://developers.arcgis.com/calcite-design-system/foundations/accessibility) for guidance.

### Summary

We got [an accessibility violation](https://zentopia.esri.com/workspaces/sprint-board-60340b9721bc3a4ab6367db5/issues/gh/dc/hub/11274) on the calcite-text-area when LevelAccess audited Hub. The `textarea` has a `aria-describedby` added to it referencing calcites out of the box validation before the validation span has even been added to the dom. It would be preferred if we could conditionally add the `aria-describedby` based on whether or not that span is added to the dom.

### Actual Behavior

Issue can be [seen on Calcite docs](https://developers.arcgis.com/calcite-design-system/components/text-area/)

In a calcite area with no validation (no max/min length) the `aria-describedby` is referencing a nonexistent span
![Image](https://github.com/user-attachments/assets/f1ddc372-1a4f-4412-a21e-6285702967ed)

In a calcite area with validation (max/min length), but not violating that validation the `aria-describedby` is referencing a nonexistent span
![Image](https://github.com/user-attachments/assets/4547b290-3bdc-4a57-8f2b-f79b2b118c43)

When the character count is gone over, the span now is present in the dom
![Image](https://github.com/user-attachments/assets/1dc2a267-5c16-46d4-b721-105310b81673)

### Expected Behavior

The `textarea` would not have an `aria-describedby` until the validation fails and the span is added to the dom, at which point then the `aria-describedby` would be added.

### Reproduction Sample

https://developers.arcgis.com/calcite-design-system/components/text-area/

### Reproduction Steps

1. visit https://developers.arcgis.com/calcite-design-system/components/text-area/ 
2. Inspect the textarea and note the `aria-describedby`
3. configure max/min length
4. Observe the span being added/removed but the `aria-describedby` remaining

### Reproduction Version

3.03

### Priority impact

impact - p3 - not time sensitive

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Hub
